source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/HorizontalFlowLayout.java

```diff
@@ -232,4 +246,44 @@ int measureRequiredHeight(final int width,
 
         return ypos;
     }
+
+    /**
+     * Measures the width required by this view
+     * by summing the width and padding of all of it's children.
+     * <p>
+     * (Package-private visibility for uint tests access.)
+     *
+     * @param paddingLeft left padding for this view.
+     * @param paddingRight right padding for this view.
+     * @return the height required by this view.
+     */
+    int measureRequiredWidth(final int paddingLeft,
+                             final int paddingRight) {
+        int requiredWidth = paddingLeft + paddingRight;
+        int childWidth, childMarginLeft, childMarginRight;
+        View child;
+        MarginLayoutParams childMarginLayoutParams;
+        // Go through the children to get the total width of the view
+        for (int i = 0; i < getChildCount(); i++) {
+            child = getChildAt(i);
+
+            if (child.getVisibility() != GONE) {
+                childWidth = child.getMeasuredWidth();
```

Comment:
if child visibility is GONE, childWidth will use the previous child's width or undefined if it's first element.

Reply:
Yes, but we only add `childWidth` to `requiredWidth` within this conditional, so we shouldn't run into any issues here?

Reply:
sorry about that. LN270 looked like it matched with LN279. This still applies if child layout param is null or is not instance of MarginLayoutParam.